napari/_vispy/vispy_shapes_layer.py

```diff
@@ -32,21 +32,27 @@ def _on_data_change(self):
         if vertices is not None:
             vertices = vertices + 0.5
 
-        if len(faces) == 0:
-            self.node._subvisuals[3].set_data(vertices=None, faces=None)
-        else:
-            self.node._subvisuals[3].set_data(
-                vertices=vertices[:, ::-1], faces=faces, face_colors=colors
-            )
+        if len(vertices) == 0 or len(faces) == 0:
+            vertices = np.zeros((3, self.layer.dims.ndisplay))
+            faces = np.array([[0, 1, 2]])
+            colors = np.array([[0, 0, 0, 0]])
+
+        self.node._subvisuals[0].set_data(
+            vertices=vertices[:, ::-1], faces=faces, face_colors=colors
+        )
```

Comment:
As usual: can you add a comment that the `::-1` converts numpy-style coordinates to xyz?

Reply:
added